packages/starlight/utils/git.ts

```diff
@@ -39,3 +40,76 @@ export function listGitTrackedFiles(directory: string): string[] {
 	const output = result.stdout.trim();
 	return output.split('\n').map((file) => join(directory, file));
 }
+
+function getRepoRoot(directory: string): string {
+	const result = spawnSync('git', ['rev-parse', '--show-toplevel'], {
+		cwd: directory,
+		encoding: 'utf-8',
+	});
+
+	if (result.error) {
+		return directory;
+	}
+
+	try {
+		return realpathSync(result.stdout.trim());
+	} catch {
+		return directory;
+	}
+}
+
+export function getAllNewestCommitDate(directory: string): [string, number][] {
```

Comment:
We should add a third changeset I think to mention this new performance improvement as this would benefit to all type of projects?